### PR TITLE
Add comprehensive stress tests for class inheritance and field handling

### PR DESCRIPTION
**Pull request by stevan**
Add t/class/stress.t with 74 tests exercising pathological combinations of the class system, with a focus on verifying that the seal-ordering fixes for GH#20890 and GH#21221 cover a wide range of edge cases.

Tests use fresh_perl_is for scenarios where a segfault is plausible (seal ordering, deep chains, eval-based class definitions) so that a crash is detected cleanly rather than taking down the test harness.

Categories and coverage:

  Seal ordering (GH#20890/GH#21221)    11 tests
    - unit-syntax base with unit/block children
    - multiple subclasses of the same unsealed base
    - mixed unit/block siblings, alternating syntax in chains
    - block-scoped child defined inside base class body

  Deep inheritance chains                4 tests
    - 5-level unit-syntax chain
    - 4-level nested block classes
    - tree-shaped (non-linear) inheritance
    - chain of empty (fieldless) classes

  Field index correctness                5 tests
    - 10+ fields in base with child additions
    - same-named fields in parent and child (shadowing)
    - fieldless children of field-heavy base
    - 50 fields in base with child field

  :param, :reader, :writer               5 tests
    - :param through 3-level chain
    - :reader and :writer across inheritance
    - duplicate :param name detection

  ADJUST blocks                          4 tests
    - multiple ADJUST blocks, ordering across chain
    - ADJUST-only subclass (no own fields)
    - exception in ADJUST during construction

  __CLASS__, SUPER, forward-declared      5 tests
    - __CLASS__ in field initializers across inheritance
    - SUPER::method chain with fields at each level
    - forward-declared methods with fields

  Closures and lexical methods            5 tests
    - anonymous method closures over fields
    - sub closures capturing fields in child instances
    - lexical methods with ->& syntax

  Safety checks                          3 tests
    - cross-class method dispatch (clean error)
    - bless into class (forbidden)
    - @ISA mutation (read-only)

  Object lifecycle                       8 tests
    - weakrefs to class instances
    - class objects as field values
    - circular references between instances
    - DESTROY with SUPER chain

  Compilation boundaries                 5 tests
    - multi-level inheritance across eval blocks
    - class defined in BEGIN, subclassed at runtime
    - class defined inside method via eval
    - two independent chains in same scope
    - block class inside unit class scope

  Stress                                 1 test
    - 1000 rapid object create/destroy cycles

  Other                                  4 tests
    - method override with own field access
    - SUPER in fieldless child of field-heavy base
    - field init order preserved across inheritance
    - method override returning child field

### CODE REVIEW

**Comment by stevan** [comment]
This covers a TON of possible segfault cases, good news, they all pass!

**Comment by mauke** [comment]
I am opposed to merging this code. It's LLM slop (`Co-Authored-By: Claude Opus 4.6 <noreply@anthropic.com>`).

**Comment by stevan** [comment]
So the CI tests are failing mostly because of porting/manifest tests (because I added a new test file).

I have to say, based on my experience chasing this kind of stuff down in the last branch with the roles tests, this is making it really difficult to write new tests. 

This is something I suggest you consider as a problem, specifically because @leonerd has been asking for tests from people for a long time.

**Comment by stevan** [comment]
> I am opposed to merging this code. It's LLM slop (`Co-Authored-By: Claude Opus 4.6 <noreply@anthropic.com>`).

Hmm, same message as before ... I am starting to suspect you might be automating this with AI or something ;)

**Comment by stevan** [comment]
I have looked over all the failing jobs and they all fail on `porting/manifest.t`, so let me first address that.

As I mentioned above, the fact that adding a new test file to the t/ directory can break the build is a pretty sizable barrier to entry. And it breaks the build for a porting test that checks the manifest, which is a very important test to run before the release, but here it seems to just be causing friction.

Additionally, and I cannot stress this enough ... **@leonerd has been asking for people to give him test files for a long time, and this branch is exactly what he has been asking for!**

Perhaps there is a different way I am supposed to go about this? If so, then this should probably be advertised more prominently. Perhaps there is some simple script which prevents this and automates things, ... also should be advertised better. 

My goal is to help get the class system ready for real world production use. I have found the feature to be very solid and reliable, **except** for the segfaults which @leonerd patch has fixed and these tests are to ensure they will not return.

I have been using the class system for the last 2+ years, pretty much since the feature was released. These segfaults were the primary reason why I have not released any of my code to CPAN and why I have not tried to push any class system adoption at work. 

---

Here is a summary of the other test failures, since i do not have a windows machine I can not test these, but there is no reason why my test would have broken these tests. 

- Windows                                              
    - ../dist/Net-Ping/t/450_service.t  TODO passed:   9
    - ../ext/IPC-Open3/t/IPC-Open3.t    TODO passed:   33

- Cygwin
    - op/sprintf2.t                       TODO passed:   1704
    - re/pat.t                            TODO passed:   1149
    - re/pat_thr.t                        TODO passed:   1149
    - ../cpan/IO-Socket-IP/t/11sockopts.t TODO passed:   1-2
    - ../dist/Net-Ping/t/450_service.t    TODO passed:   9

As for this run/todo.t tests, I couldn't figure this one out, the logs were not clear. 

- linux (-Duseshrplib -Dusequadmath -Dusecbacktrace -Dusethreads)
- install
    - run/todo.t

**Comment by stevan** [comment]
This push *should* fix the porting/manifest test, but the Windows and Cygwin issues are not related to this at all.

Additionally, I dug around and found the documentation on the MANIFEST test, it is in Porting/pumpkin.pod and in the release managers guide. Neither one calls out test files as a specific issue, though that is not really a big deal as it is not a common thing to do. That said, if you are actively soliciting test cases for this feature, you should consider making this easier to do, or being specific about how you want people to go about it.

**Comment by stevan** [comment]
Excellent, looks like everything is green now. 

I assume we can merge this? unless someone has technical objections to the tests themselves.

**Comment by stevan** [comment]
I have added more tests to increase the coverage of edge cases in t/class, the summary about what is tested are in the commit message, but I will repeat it here to make it easier if anyone wants to discuss.

---

- errors.t

Error messages: bless-into-class, reopen sealed class, read-only @isa,
cross-class method, non-instance method, unrecognized/missing params, odd args,
duplicate :param, :writer on non-scalar, invalid accessor names,
incomplete class, CLASS misuse, field access violations, :isa with non-class,
multiple :isa, unrecognized attributes, pre-existing @isa

- attribute_combos.t

All combinations of :param/:reader/:writer with default and custom names,
//= and ||= with :reader, multi-field combos, combos across inheritance,
array/hash :reader, writer chaining, copy semantics

- module.t

Multi-class evals, regular subs alongside methods, use constant, use overload,
package variables, eval across compilation units, BEGIN blocks, nested package
names, Exporter-like import, AUTOLOAD, isa/DOES/can, BEGIN+ADJUST coexistence

- edge_cases.t

Empty/fields-only/methods-only/ADJUST-only classes, deep package names, method
chaining, self-calls, wantarray, caller(), virtual dispatch, independence of
instances, ADJUST die recovery, eval in methods, 100 fields, chained field
init, nested classes, paramless constructor, object identity

- lifecycle.t

Construction order (fields-then-ADJUSTs across inheritance), DESTROY ordering,
field destruction order, weak references, destruction during errors, ADJUST
die + DESTROY, refcount verification, objects in containers, rapid
create/destroy stress

- interop.t

blessed/ref/reftype/refaddr, isa (method + operator + UNIVERSAL), DOES, can,
stringify/numify, comparison ops, objects in data structures, eval/die
interaction, local(), Scalar::Util, traditional vs modern OO coexistence,
:isa rejects non-class, MRO, for loops, boolean context, sprintf

**Comment by stevan** [comment]
If the additional tests are too much to review in one branch, I am happy to split them out into separate PRs if desired.

File: t/class/stress.t [hunk 1]
```diff
@@ -0,0 +1,942 @@
...
+    class TA; field $a = "a"; method a { $a }
+    class TB :isa(TA); field $b = "b"; method b { $b }
+    class TC :isa(TA); field $c = "c"; method c { $c }
+    class TD :isa(TB); field $d = "d"; method d { $d }
+    class TE :isa(TC); field $e = "e"; method e { $e }
+}
+my $d = TD->new;
+my $e = TE->new;
+print $d->a, $d->b, $d->d, " ", $e->a, $e->c, $e->e, "\n";
+CODE
```
Comment (richardleach) [inline t/class/stress.t:180]: What does this test that the earlier linear chain tests did not? Some kind of leakage between classes derived from the same ancestor?

Comment (stevan) [reply t/class/stress.t:180]: This tests that more extensively, with more things going on. It was not a bug for a specific issue, but an expansion on the existing test case to make sure we didn't miss something. 

Honestly, this is probably a more thorough test of the feature, so it could replace the other. But I opted to create all new tests to avoid merge conflicts with Paul's ongoing work. That said I would be happy to merge things if folks are open to it.

Comment (stevan) [reply t/class/stress.t:180]: As I said at the top of the PR, these are "pathological combinations" that are meant to push the boundaries of the segfault issue in particular. I see the segfaults as one of the big blockers for adoption and something that has personally gotten on my nerves since 5.36.

File: t/class/stress.t [hunk 2]
```diff
@@ -0,0 +1,942 @@
...
+    class ShadowChild :isa(ShadowBase) {
+        field $x = "child-x";
+        field $y = "child-y";
+        method child_x { $x }
+        method child_y { $y }
+    }
+
+    my $obj = ShadowChild->new;
+    is($obj->base_x, "base-x", 'Parent method sees parent field $x');
+    is($obj->base_y, "base-y", 'Parent method sees parent field $y');
```
Comment (richardleach) [inline t/class/stress.t:255]: _perlclass.pod_ says `Inherited methods are visible and may be invoked. Fields are always lexical and therefore not visible by inheritance.`

Does that statement, the implemented behaviour being tested, and the names of these tests all match up? (I'm honestly not sure either way.)

**Comment by stevan** [comment]
I mentioned this in the comment reply to @richardleach  but it is worth repeating here as well.

I purposefully opted to create all new test files in this PR to avoid merge conflicts with Paul's ongoing work. My previous (crazy) roles PR had this problem when Paul committed the fix for the segfaults into blead, effectively making my branch unmergable (which yes, you could argue it was already, but that is a different discussion).  

In order to avoid this happening again, or getting in the way of any of Paul's work, I chose this approach. If preferred I can merge tests together with existing ones, but for now I opted for a less disruptive path.

File: t/class/stress.t (hunk 2 above)
Comment (stevan) [reply t/class/stress.t:255]: This is testing the accessors and not the fields. Accessors **are** visible via inheritance, giving access to the lexical field's value through them.

**Comment by stevan** [comment]
I understand that – at least at the moment – contributions made by AI/LLMs are considered suspect and risky, and that there needs to be more discussions on this topic before PRs like this can be merged. 

I want to be clear that I respect this position and understand the complexities of it, so I am closing this PR.
